ICP-10783, Fix to handle notification report for Dawon switch on/off

lukhong opened the pull request:
WWST-5338

lukhong committed:
ICP-10783, Fix to handle notification report for Dawon switch on/off

Signed-off-by: Ki-wook <kiwook.hong@samsung.com>

greens commented:
Only issue I have with this is that it makes the `runLocally` untrue. @kianooshST : thoughts? This device for some reason chose not to use any of the switch command classes to report its state.

lukhong commented:
@kianooshST Do you have any opinion on this?

kianooshkarami commented on devicetypes/smartthings/zwave-metering-switch.src/zwave-metering-switch.groovy:
```diff
@@ -199,6 +199,16 @@ def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerS
 
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd){
+	if ((cmd.notificationType == 0x08) && zwaveInfo?.mfr?.equals("018C")) {
+		if (cmd.event == 0x02) {
```
Looking at the Notification Type of Power Management (0x08) and the following state/events:

0x02: AC Mains Disconnected
0x03: AC Mains re-connected

I think this makes sense. It would be great to add a comment in the code.

The only think I would ask is that, does it makes sense to have this  for all devices that have "018C" manufacturer or just do it for this particular device.

lukhong replied:
This Korean company is just started to work with SmartThings. Currently requested devices that have a switch seem to have the same approach. So I can say having 018C mfr in the code makes sense currently.

kianooshkarami commented:
> Only issue I have with this is that it makes the runLocally untrue

because of this reason, I would suggest to create a new DTH just for this device. @greens Thoughts?

greens commented:
I find it frustrating to make a whole new DTH for such a little thing, but I guess we should commit to actually be running locally when we say we are.

@lukhong can you instead make a new PR that is a copy of `zwave-metering-switch`, but with these changes and without the already-existing fingerprints?

kianooshkarami commented:
@greens I do not disagree with your assessment about a new DTH that is almost the same.

lukhong commented:
Of course, I can create a new DTH for this mfr's devices and will request a review.

lukhong committed:
Revert "ICP-10783, Fix to handle notification report for Dawon switch on/off"

This reverts commit 8459b303c540f662a74503328ab55255296ec254.

lukhong committed:
Create a new DTH for Dawon devices

Signed-off-by: Ki-wook <kiwook.hong@samsung.com>

lukhong commented:
@greens I've created a new DTH for Dawon device.
- Removed some codes existed for other mfr's.
- Changed name, added vid to use the existing ui-metadata of Z-Wave metering switch.
- Removed FPs in the generic DTH.
